test/reference-implementations.test.ts

```diff
@@ -184,32 +191,136 @@ describe('reference implementation tests', () => {
     // We only test the BIP-32 vectors with deriveKeyFromPath, since not all
     // paths are BIP-44 compatible.
     describe('deriveKeyFromPath', () => {
-      it('derives the test vector keys', () => {
-        vectors.forEach((vector) => {
+      it('derives the test vector keys', async () => {
+        for (const vector of vectors) {
           const seed = hexStringToBuffer(vector.hexSeed);
           const seedKey = createBip39KeyFromSeed(seed);
 
-          vector.keys.forEach(
-            (keyObj: {
-              path: any;
-              extPrivKey: string;
-              extPubKey: string;
-              privateKey: string;
-            }) => {
-              const { path, privateKey } = keyObj;
+          for (const keyObj of vector.keys) {
+            const { path, privateKey } = keyObj;
+
+            let targetKey: Buffer;
+            if (path.ours.string === '') {
+              targetKey = seedKey;
+            } else {
+              targetKey = await deriveKeyFromPath(
+                path.ours.tuple as HDPathTuple,
+                seedKey,
+              );
+            }
+
+            expect(targetKey.slice(0, 32).toString('hex')).toStrictEqual(
+              privateKey,
+            );
+          }
+        }
+      });
+    });
+  });
+
+  describe('ed25519', () => {
+    describe('SLIP-10', () => {
+      const vectors = fixtures.ed25519.slip10;
+
+      describe('deriveKeyFromPath', () => {
+        it('derives the test vector keys', async () => {
+          for (const { hexSeed, keys } of vectors) {
+            const seedKey = createBip39KeyFromSeed(
+              hexStringToBuffer(hexSeed),
+              ed25519,
+            );
 
+            for (const { path, privateKey, publicKey } of keys) {
               let targetKey: Buffer;
               if (path.ours.string === '') {
                 targetKey = seedKey;
               } else {
-                targetKey = deriveKeyFromPath(path.ours.tuple, seedKey);
+                targetKey = await deriveKeyFromPath(
+                  path.ours.tuple as HDPathTuple,
+                  seedKey,
+                  path.ours.tuple.length,
+                  ed25519,
+                );
               }
 
-              expect(targetKey.slice(0, 32).toString('hex')).toStrictEqual(
-                privateKey,
-              );
-            },
+              expect(targetKey.slice(0, 32).toString('hex')).toBe(privateKey);
+              expect(
+                (await ed25519.getPublicKey(targetKey.slice(0, 32))).toString(
```

Comment:
Is there a reason `getPublicKey` couldn't do the slicing here?

Reply:
`ed25519.getPublicKey` takes a private key, but `targetKey` is a buffer with the private key and chain code appended to it. I don't think it makes sense to handle slicing in the `getPublicKey` function here.